app/src/main/kotlin/io/treehouses/remote/Fragments/DiscoverFragment.kt

```diff
@@ -0,0 +1,232 @@
+        bind = ActivityDiscoverFragmentBinding.inflate(inflater, container, false)
+        deviceList.clear()
+        mChatService = listener.getChatService()
+        mChatService.updateHandler(mHandler)
+        requestNetworkList()
+        requestGatewayInfo()
+        return bind.root
+    }
+
+    private fun setupIcons() {
+        bind.container.removeAllViewsInLayout()
+
+        val midX = (bind.container.measuredWidth / 2).toFloat()
+        val midY = (bind.container.measuredHeight / 2).toFloat()
+
+        val r = this.resources.displayMetrics.widthPixels / 2 * 0.72
+        val interval = 2 * PI / (deviceList.size - 1)
+        var radians = Random.nextFloat() * 2 * PI
+
+        var size: Int = when {
+            deviceList.size <= 12 -> ICON_MEDIUM_SIZE
+            deviceList.size <= 20 -> ICON_SMALL_SIZE
+            else -> ICON_XSMALL_SIZE
+        }
+
+        for(idx in 1 until deviceList.size) {
+            val i = deviceList[idx]
+            val imageView = ImageView(context)
+            imageView.setImageResource(R.drawable.circle_yellow)
+            imageView.layoutParams = LinearLayout.LayoutParams(size, size)
+            imageView.x = midX + (r * sin(radians)).toFloat() - size / 2
+            imageView.y = midY + (r * -cos(radians)).toFloat() - size / 2
+
+            imageView.setOnClickListener {
+                val message = ("IP Address: " + i.ip + "\n") +
+                        ("MAC Address: " + i.mac)
+                message.lines()
+                showDialog("Device Information", message)
+            }
+
+            val bitmap = Bitmap.createBitmap(
+                bind.container.measuredWidth,
+                bind.container.measuredHeight,
+                Bitmap.Config.ARGB_8888
+            )
+
+            val canvas = Canvas(bitmap)
+            val p4 = Paint()
+            p4.isAntiAlias = true
+            p4.color = Color.BLACK
+            p4.strokeWidth = 10f
+            canvas.drawLine(midX, midY, imageView.x + size / 2, imageView.y + size / 2, p4)
+
+            radians = (radians + interval) % (2 * PI)
+
+            val line = ImageView(context)
+            line.setImageBitmap(bitmap)
+            bind.container.addView(line)
+            bind.container.addView(imageView)
+        }
+
+        size = ICON_MEDIUM_SIZE
+
+        val icon = ImageView(context)
+        icon.setImageResource(R.drawable.ic_bluetooth)
+        val param = LinearLayout.LayoutParams(size, size)
+
+        icon.x = midX - size / 2
+        icon.y = midY - size / 2
+
+        icon.layoutParams = param
+        gatewayIcon = icon
+
+//        bind.container.addView(icon)
+    }
+
+    private fun showDialog(title: String, message: String) {
+        val alertDialog = CreateAlertDialog(context, R.style.CustomAlertDialogStyle,title,message)
+                .setPositiveButton("OK") { dialog: DialogInterface, _: Int -> dialog.dismiss() }.create()
+        alertDialog.show()
```

Comment:
Working for me, adding this would fix the alert dialog corners in dark mode and make them rounded

```suggestion
        alertDialog.window!!.setBackgroundDrawableResource(android.R.color.transparent)
        alertDialog.show()
```